### PR TITLE
OpenShift deploy, start & create_provider

### PR DESCRIPTION
**Pull request by dkorn**
- made changes in openshift deploy script
- added the create_provider task in manageiq module
- added openshift start script, that includes adding openshift
  as a provider in manageiq.

### CODE REVIEW

File: projects/manageiq.py [hunk 1]
```diff
@@ -96,3 +98,23 @@ def start():
...
+        "action": "create",
+        "resources": [
+            {
+                'name': name,
+                'type': 'EmsKubernetes',
+                'hostname': hostname,
+                'port': '8443'
+            }]
+        }
+    data = json.dumps(body)
```
Comment (simon3z) [inline projects/manageiq.py]: Can be simplified with:

```
data = json.dumps({
    "action": "create",
    "resources": [
        {
            'name': name,
            'type': 'EmsKubernetes',
            'hostname': hostname,
            'port': '8443'
        }]
    })
```

File: projects/manageiq.py [hunk 2]
```diff
@@ -96,3 +98,23 @@ def start():
...
+
+    manageiq_url = 'http://{0}:3000/api/providers/'.format(hostname)
+    r = requests.post(manageiq_url, data, auth=(credentials['user'],
+                                                credentials['password']))
+    print r.json
```
Comment (simon3z) [inline projects/manageiq.py:129]: Is this a leftover or it's for debugging purpose?

File: projects/openshift.py [hunk 3]
```diff
@@ -35,20 +37,37 @@
...
+def deploy():
     install_deps()
 
-    enable_services(['docker.socket'])
+    enable_services(['docker.service'])
+
     enable_tcp_ports([8443, 8444])
 
-    sudo('usermod -a -G docker {0}'.format(env.user))
+    sudo('groupadd docker')
```
Comment (simon3z) [inline projects/openshift.py]: This shouldn't be required. When you install docker it should also add a docker group.

File: projects/openshift.py [hunk 4]
```diff
@@ -35,20 +37,37 @@
...
+
+@task
+def start():
+    with cd('origin'):
+        run('make')
+        run('nohup ./_output/local/go/bin/openshift start &')
+
+        # add OpenShift as a provider in ManageIQ
+        create_provider('openshift01', env.host, {'user': 'admin',
+                                                  'password': 'smartvm'})
```
Comment (simon3z) [inline projects/openshift.py]: Make these (admin/smartvm) defaults of `start` parameters. Even better if you move this to a different task.

File: projects/manageiq.py (hunk 1 above)
Comment (dkorn) [inline projects/manageiq.py]: Done.

File: projects/manageiq.py (hunk 2 above)
Comment (dkorn) [inline projects/manageiq.py:129]: for debugging purpose, tell me if you think it's needless.

File: projects/openshift.py (hunk 3 above)
Comment (dkorn) [inline projects/openshift.py]: I thought so too, but it's not added automatically and the script fails without it:
'usermod: group 'docker' does not exist'

File: projects/openshift.py (hunk 4 above)
Comment (dkorn) [inline projects/openshift.py]: Done.

File: projects/openshift.py (hunk 3 above)
Comment (simon3z) [inline projects/openshift.py]: What are the permissions of your docker socket:

```
ls -l /var/run/docker.sock 
srw-rw----. 1 root docker 0 Apr 28 10:11 /var/run/docker.sock
```

Comment (dkorn) [inline projects/openshift.py]: srw-rw----. 1 root docker 0 Apr 22 04:21 /var/run/docker.sock

Comment (dkorn) [inline projects/openshift.py]: we agreed on: "sudo('usermod -aG root {0}'.format(env.user))"
so that the user will be able to use docker without sudo.

**Pull request merged**